game/src/ecs/entities/Entity.java

```diff
@@ -11,16 +11,12 @@
 /** Entity is a unique identifier for an object in the game world */
 @DSLType(name = "game_object")
 @DSLContextPush(name = "entity")
-public class Entity {
+public record Entity(HashMap<Class, Component> components, Logger entityLogger, int id) {
     private static int nextId = 0;
-    public final int id = nextId++;
-    private HashMap<Class, Component> components;
-    private final Logger entityLogger;
 
     public Entity() {
-        components = new HashMap<>();
+        this(new HashMap<>(), Logger.getLogger(Entity.class.getName()), nextId++);
         Game.addEntity(this);
```

Comment:
Dumme Frage I: Wir hätten jetzt einen `public` Ctor mit allen Parametern, den man auch aufrufen kann. Damit
- könnte jemand frei eine beliebige `id` mitgeben (vermutlich nicht schlimm, siehe II)
- würde die neue Entität nicht in `Game` registriert.

Dumme Frage II: Der Parameter `int id` wird irgendwie gar nicht genutzt. Könnte der dann nicht aus der Schnittstelle raus?

Dumme Frage III (analog zu II): Der Parameter `Logger entityLogger` wird irgendwie gar nicht genutzt. Könnte der dann nicht aus der Schnittstelle raus?

Dumme Frage IV (als Folge aus I): Sollte man den Record ohne Parameter definieren? Für die `nextID` muss ja sowieso schon ein Attribut manuell angelegt werden, dann könnte man auch `components` und `entityLogger` so handhaben? Dazu kommt ja auch, dass Java sonst je einen Getter basteln würde (und die will man ja eigentlich nicht?) - Aber dann bräuchte das Ding auch kein Record sein?

Reply:
Ich habe das ganze jetzt zu einer `final class` gemacht und damit die ganzen Probleme gelöst. 

Die id lass ich erstmal drinnen, sollte aber nochmal drüber nachgedacht werden.

Reply:
> Gegenfrage: Warum hatten wir uns eigentlich so auf Record eingeschoßen?
> zu einer final class gemacht 

Weiss ich auch nicht, vermutlich ist das so ein Über-Generalisierungs-Ding ... Ich glaube, `final class` passt hier deutlich besser :)